### PR TITLE
Create reviews

### PR DESCRIPTION
**Pull request by Bryan-Genest**
1. User can add review to beach show page
2. Index page has links to beaches show page
3. Beach and Reviews now serializes, one fetch for Beach show page
4. Review json tests (structure of response)
5. Refactored ReviewShowTile.js, moved in logic from parent file BeachesShow.js

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -6,11 +6,12 @@
 
   namespace :api do
     namespace :v1 do
-      resources :beaches, only: [:index, :show]
+      resources :beaches, only: [:index, :show] do 
+        resources :reviews, only: [:index, :create]
```
Comment (nwalberts) [inline config/routes.rb:10]: remove index

File: db/seeds.rb [hunk 2]
```diff
@@ -49,4 +42,20 @@
...
+            title: "A beach",
+            rating: 3,
+            text: "This is a review",
+            beach_id: beach1.id
+)
+
+review2 = Review.create(
+            title: "A beach 2",
+            rating: 1,
+            text: "This is a bad review",
```
Comment (nwalberts) [inline db/seeds.rb:57]: i know its annoying, but can we have slightly more real review text?

File: db/seeds.rb [hunk 3]
```diff
@@ -49,4 +42,20 @@
...
+            beach_id: beach1.id
+)
+
+admin1 = User.create!({:email => "myemail@gmail.com", :role => "admin", :password => "111111", :password_confirmation => "111111" })
```
Comment (nwalberts) [inline db/seeds.rb:61]: we should not use hashrocket syntax. use the syntax from the rest of the seed file!

File: spec/features/user_creates_new_review.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      post :create, params: {review: post_json, beach_id: first_beach.id}, format: :json
+      returned_json = JSON.parse(response.body)
+
+      expect(response.status).to eq 200
+      expect(response.content_type).to eq("application/json")
+
+      expect(returned_json["title"]).to eq first_review.title
+      expect(returned_json["rating"]).to eq first_review.rating
+      expect(returned_json["text"]).to eq first_review.text
+
```
Comment (nwalberts) [inline spec/features/user_creates_new_review.rb:23]: we need a test for the sad path as well!

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: db/seeds.rb (hunk 2 above)
Comment (JulianBaxandall) [reply db/seeds.rb:57]: -This is a bad review of this suggestion-

**Comment by Bryan-Genest** [comment]
Why does git say this has unmerged changes 🧐🧐

**Pull request closed without merging**